extensions/wikia/TemplateClassification/specials/TemplatesSpecialController.class.php

```diff
@@ -0,0 +1,335 @@
+			$template['wlh'] = SpecialPage::getTitleFor( 'Whatlinkshere', $title->getPrefixedText() )->getLocalURL();
+			$template['revision'] = $this->getRevisionData( $title );
+		}
+
+		return $template;
+	}
+
+	/**
+	 * Get data about tempalte page last revision
+	 *
+	 * @param Title $title
+	 * @return array
+	 * @throws MWException
+	 */
+	private function getRevisionData( Title $title ) {
+		$data = [];
+		$revision = Revision::newFromId( $title->getLatestRevID() );
+
+		if ( $revision instanceof Revision ) {
+			$data['timestamp'] = wfTimestamp( TS_UNIX, $revision->getTimestamp() );
+
+			$user = $revision->getUserText();
+
+			if ( $revision->getUser() ) {
+				$userpage = Title::newFromText( $user, NS_USER )->getFullURL();
+			} else {
+				$userpage = SpecialPage::getTitleFor( 'Contributions', $user )->getFullUrl();
+			}
+
+			$data['username'] = $user;
+			$data['userpage'] = $userpage;
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Prepare pagination
+	 *
+	 * @param int $total
+	 * @param int $page
+	 */
+	private function preparePagination( $total, $page ) {
+		$itemsPerPage = self::ITEMS_PER_PAGE;
+		$params = [ 'page' => '%s' ];
+
+		if ( $this->type ) {
+			$params['type'] = $this->type;
+		}
+
+		if ( $this->templateName ) {
+			$params['template'] = $this->templateName;
+		}
+
+		if( $total > $itemsPerPage ) {
+			$paginator = Paginator::newFromArray( array_fill( 0, $total, '' ), $itemsPerPage, 3, false, '',  self::ITEMS_PER_PAGE );
+			$paginator->setActivePage( $page );
+			$url = urldecode( $this->specialPage->getTitle()->getLocalUrl( $params ) );
+			$this->paginatorBar = $paginator->getBarHTML( $url );
+		}
+	}
+
+	/**
+	 * Slice templates list for pagination
+	 *
+	 * @param int $page
+	 */
+	private function sliceTemplates( $page ) {
+		$offset = $page * self::ITEMS_PER_PAGE;
+		$limit = self::ITEMS_PER_PAGE;
+
+		$this->templates = array_slice( $this->templates, $offset, $limit, true );
+	}
+
+	/**
+	 * Get all existing and user facing template types on wiki
+	 *
+	 * @param array $classifiedTemplates
+	 * @return array
+	 */
+	private function getTemplateGroups( $classifiedTemplates ) {
+		$groups = array_unique( $classifiedTemplates );
+
+		foreach ( $groups as $id => $group ) {
+			if ( !$this->isUserType( $group ) ) {
+				if ( !in_array( TemplateClassificationService::TEMPLATE_UNKNOWN, $groups ) ) {
+					$groups[] = TemplateClassificationService::TEMPLATE_UNKNOWN;
+				}
+
+				unset( $groups[$id] );
+			}
+		}
+
+		sort( $groups );
+
+		return $groups;
+	}
+
+	/**
+	 * Get number of templates
+	 *
+	 * @return int
+	 */
+	private function getTotalTemplatesNum() {
+		return count( $this->templates );
+	}
+
+	/**
+	 * Get template type
+	 *
+	 * @param int $pageId
+	 * @param array $classifiedTemplates
+	 * @return string
+	 */
+	private function getTemplateType( $pageId, $classifiedTemplates ) {
+		if ( !isset( $classifiedTemplates[$pageId] ) || !$this->isUserType( $classifiedTemplates[$pageId] )
+		) {
```

Comment:
I'd consider splitting the conditions over two lines, rather than just the closing brackets.

Also, for this and the `getDefaultType` method below, the `else` block is unnecessary since the `if` has a return, so I think it looks cleaner as:
```php
private function getTemplateType( $pageId, $classifiedTemplates ) {
	if ( !isset( $classifiedTemplates[$pageId] ) || !$this->isUserType( $classifiedTemplates[$pageId] ) ) {
		return TemplateClassificationService::TEMPLATE_UNKNOWN;
	}

	return $classifiedTemplates[$pageId];
}
```